### PR TITLE
Update gradle to 4.1 and switch build script to Kotlin.

### PR DESCRIPTION
**Pull request by PeterJohnson**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+                    .emacs
+                    .destination
+                    .takeIf { it.exists() }
+                    ?.readText()
+                    .takeIf { !it.isNullOrBlank() }
+                    ?.also { logger.warn(it) }
+        }
+    })
+}
+*/
```
Comment (JLLeitschuh) [inline build.gradle.kts]: What's wrong with this giant commented out chunk?

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        classpath("org.junit.platform:junit-platform-gradle-plugin:1.0.0")
+    }
+}
+plugins {
+    application
+    `maven-publish`
+    jacoco
+    java
+    //checkstyle
+    //pmd
```
Comment (JLLeitschuh) [inline build.gradle.kts]: Favor this style over the apply syntax.

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+ * Retrieves the [pmd][org.gradle.api.plugins.quality.PmdExtension] project extension.
+ */
+val Project.`pmd`: org.gradle.api.plugins.quality.PmdExtension get() =
+    extensions.getByName("pmd") as org.gradle.api.plugins.quality.PmdExtension
+
+/**
+ * Configures the [pmd][org.gradle.api.plugins.quality.PmdExtension] project extension.
+ */
+fun Project.`pmd`(configure: org.gradle.api.plugins.quality.PmdExtension.() -> Unit) =
+        extensions.configure("pmd", configure)
```
Comment (JLLeitschuh) [inline build.gradle.kts]: Using the `plugins` syntax you shouldn't need a lot of this.

File: build.gradle.kts (hunk 1 above)
Comment (PeterJohnson) [reply build.gradle.kts]: build fails spectacularly because the code doesn't pass checkstyle or pmd

File: build.gradle.kts (hunk 3 above)
Comment (PeterJohnson) [reply build.gradle.kts]: this was copy and paste from OutlineViewer's script..

**Comment by sciencewhiz** [comment]
Fixes #33 and #9

**Comment by PeterJohnson** [comment]
No longer required thanks to #70.

**Pull request closed without merging**